R/estimateAlpha.R

```
@@ -0,0 +1,274 @@
+                      # cleaning index from suffix to be used with the corresponding index
+                      # function
+                      index[i] <- gsub("_diversity", "", index[i])
+                      FUN <- .estimate_diversity
+                  } else if (any(grepl(index[i], .get_indices("dominance")))) {
+                      name[i] <-  .parse_name(index[i], name[i], "dominance")
+                      index[i] <- gsub("_dominance", "", index[i])
+                      FUN <- .estimate_dominance
+                  } else if (any(grepl(index[i], .get_indices("evenness")))) {
+                      name[i] <- .parse_name(index[i], name[i], "evenness")
+                      if (index[i]!="simpson_evenness") {
+                          index[i] <- gsub("_evenness", "", index[i])
+                          }
+                      FUN <- .estimate_evenness
+                  } else if (any(grepl(index[i], .get_indices("richness")))) {
+                      name[i] <- .parse_name(index[i], name[i], "richness")
+                      index[i] <- gsub("_richness", "", index[i])
+                      FUN <- .estimate_richness
+                  } else {
+                  stop("'index' is coresponding to none of the alpha diversity indices.
+                  'index' should be one of: coverage_diversity, fisher_diversity,
+                  faith_diversity, gini_simpson_diversity,
+                  inverse_simpson_diversity, log_modulo_skewness_diversity,
+                  shannon_diversity, absolute_dominance, dbp_dominance,
+                  core_abundance_dominance, gini_dominance,
+                  dmn_dominance, relative_dominance, simpson_lambda_dominance,
+                  camargo_evenness, pielou_evenness, simpson_evenness,
+                  evar_evenness, bulla_evenness, ace_richness, chao1_richness,
+                  hill_richness or observed_richness.",
+                       call. = FALSE)
+                  }
```

Check already in input check that every index can be found. There you can also parse the name. --> fail fast